### ISSUE TITLE
Document vim setup

### ISSUE THREAD

**Issue by hcarty** [ocaml-ppx/ocamlformat#38]
It would be nice to have documentation available describing how to use ocamlformat from within vim.

**Comment by jberdine** [comment]
Agreed! I don't know vim, but I expect that a slight adaptation of [vim-reason](https://github.com/reasonml-editor/vim-reason) should work.

**Comment by hcarty** [comment]
I think vim-reason uses stdin/stdout for processing buffers, at least by default.  But it should be possible to work around the difference with some temporary file writing.

**Comment by jberdine** [comment]
Infer uses a [script](https://github.com/facebook/infer/blob/master/scripts/ocamlformat.sh) to interact with tools expecting output on stdout, but still takes input in a file. The emacs integration also [creates temp files](https://github.com/ocaml-ppx/ocamlformat/blob/master/emacs/ocamlformat.el#L168-L170). Note that the `--name` option of ocamlformat can be useful to get warning/error messages to refer to the original file instead of a temp.

**Comment by jberdine** [comment]
This might be easier after 0e02d5cb8e0f850d135d83cee027b30e11441a50.

**Comment by dhcmrlchtdj** [comment]
I would recommend [neoformat](https://github.com/sbdchd/neoformat) for vim users. :smile:

```vim
let g:neoformat_ocaml_ocamlformat = {}
let g:neoformat_ocaml_ocamlformat.exe = 'ocamlformat'
let g:neoformat_ocaml_ocamlformat.args = ['--inplace']
let g:neoformat_ocaml_ocamlformat.replace = 1
let g:neoformat_enabled_ocaml = ['ocamlformat']
```

**Comment by bobbypriambodo** [comment]
Update that neoformat now has [OCaml configuration built-in](https://github.com/sbdchd/neoformat/blob/master/autoload/neoformat/formatters/ocaml.vim) using both ocp-indent and ocamlformat if any or both installed, so users wouldn't need to manually configure it as shown by @dhcmrlchtdj above.

**Comment by pivdets** [comment]
@bobbypriambodo 

Great thanks.

![neoformat](https://user-images.githubusercontent.com/37371804/38227389-8f81c9f6-3730-11e8-85d9-146e040817c4.gif)

I just installed neoformat and using it as above, but it put all the code in one line. I was wondering if maybe this is the right way?

**Comment by hcarty** [comment]
@pivdets ocamlformat keeps code very vertically compact.  I haven't counted the characters in your example, but in general if a single expression can fit on one line (within the width limit which defaults to 80 characters) it will be formatted to one line.

**Comment by gpetiot** [comment]
Closed with #449
